### PR TITLE
Closes #912

### LINKED ISSUE #912: Add SourceData NLP Dataset
## Adding a Dataset
- **Name:** *SourceData NLP*
- **Description:** *SourceData-NLP is a named entity recognition and entity linking/disambiguation dataset produced through the routine curation of papers during the publication process. All annotations are in figure legends from published papers in molecular and cell biologyThe dataset consists of eight classes of biomedical entities (small molecules, gene products, subcellular components, cell lines, cell types, tissues, organisms, and diseases), their role in the experimental design, and the nature of the experimental method as an additional class. SourceData-NLP contains more than 620,000 annotated biomedical entities, curated from 18,689 figures in 3,223 papers in molecular and cell biology.*
- **Task:** *NER, NED*
- **Paper:**https://arxiv.org/abs/2310.20440**
- **Data:** *link to the Github repository or current dataset location*
- **License:** *CC BY 4.0*
- **Motivation:** *This is currently the largest, most detailed entity linking dataset available for scientific literature, exceeding the size of MedMentions by 2x.  Moreover, the entities linked in this dataset are annotated in figure captions, providing a dataset that is complementary to existing datasets that primarily target abstracts.  Figure captions have been shown to have the highest density of scientific of any section of biomedical research article, making this data particularly relevant to scientific information extraction.*

### PR DESCRIPTION
**Pull request by davidkartchner**
### Closes 912

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `hub/hub_repos/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio_hub <dataset_name> [--data_dir /path/to/local/data] --test_local`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: bigbio/hub/hub_repos/sourcedata_nlp/sourcedata_nlp.py [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+"""
+
+import itertools
+import json
+import os
+from typing import Dict, List, Tuple
+
+import datasets
+
+from bigbiohub import BigBioConfig, Tasks, kb_features
```
Comment (phlobo) [inline bigbio/hub/hub_repos/sourcedata_nlp/sourcedata_nlp.py:35]: Import from `.bigbiohub` needs to be relative

File: requirements.txt
Comment (phlobo) [inline requirements.txt:1]: Deleted by accident?

**Review by phlobo — COMMENTED** [review]
Hi @davidkartchner, please have a look at my comments, I cannot run the unit tests at the moment due to a wrong important. Also, I guess the requirements.txt was deleted accidentally.

File: bigbio/hub/hub_repos/sourcedata_nlp/sourcedata_nlp.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="sourcedata_nlp_source",
+            version=SOURCE_VERSION,
+            description="sourcedata_nlp source schema",
+            schema="source",
+            subset_id="sourcedata_nlp",
+        ),
+        # BigBioConfig(
```
Comment (phlobo) [inline bigbio/hub/hub_repos/sourcedata_nlp/sourcedata_nlp.py:91]: Please remove commented code

**Comment by phlobo** [comment]
@mariosaenger  I implemented the requested changes myself, could you please have a final look at the PR before merging?

**Review by mariosaenger — APPROVED** [review]
LGTM. Thanks for implementing and fixing this!

**Pull request merged**